backend/frontend/src/utils/webapi.js

```diff
@@ -114,8 +115,12 @@ var posts = new Map();
 var cached = null;
 function getPiazzaPosts(courseId: number){
   let endpoint = mainUrl + `/${courseId}/course/getpiazzaposts/`;
-  http.get(endpoint, function(res){
+  let ignoredata = false;
+  var req = http.get(endpoint, function(res){
     res.on('data', function(buf){
+      if(ignoredata){
```

Comment:
Does `ignoreData` signify if we have or don't have a valid piazza ID? If not, what is the purpose of it, at a high level?